### PR TITLE
Enabled array of object tests

### PR DESCRIPTION
**Pull request by Autophagy**
@chaudum please review

### CODE REVIEW

File: driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java [hunk 1]
```diff
@@ -129,8 +129,14 @@ private static void insertIntoTable() throws SQLException {
         preparedStatement.setObject(10, objectField);
         preparedStatement.setString(11, "1970-01-01");
         preparedStatement.setString(12, "127.0.0.1");
-        preparedStatement.setArray(13, connection.createArrayOf("varchar", new String[]{"a", "b", "c", "d"})); // TODO array support
-        preparedStatement.setArray(14, null);
```
Comment (chaudum) [inline driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java]: why did you remove this line?

**Comment by chaudum** [comment]
in order that tests are passing you need to change the upstream on your branch to point to the commit hash of your branch (https://github.com/crate/pgjdbc/pull/7/commits/d2d836a8234ec91d668d7924a43e9b3432ad9b9d)

File: driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java [hunk 2]
```diff
@@ -129,8 +129,14 @@ private static void insertIntoTable() throws SQLException {
...
+        preparedStatement.setArray(13, connection.createArrayOf("varchar", new String[]{"a", "b", "c", "d"}));
```
Comment (chaudum) [inline driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java]: this does not make sense for users, because crate does not have `varchar` datatype
I think we need to provide a crate type to psql type mapping, so users can actually use crate types when creating a `java.sql.Array`

Comment (chaudum) [inline driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java]: the usage must also be documented.
https://github.com/crate/crate-jdbc/blob/master/docs/index.txt#L209-L230

Comment (chaudum) [inline driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java]: I'd also create a separate test for setting and getting arrays of various inner types

File: driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java (hunk 1 above)
Comment (Autophagy) [inline driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java]: Ah sorry - I interpreted it as a comment todo the array tests!

File: driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java (hunk 2 above)
Comment (Autophagy) [inline driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java]: 1) Would it be best to just change the datatype strings in the info cache as a map? See here https://github.com/crate/pgjdbc/blob/crate/pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java#L70

2) Do you think this test should also be in `CrateJDBCConnectionTest` or in a separate testing class?

File: driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java (hunk 1 above)
Comment (Autophagy) [inline driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java]: Ah - I misinterpreted what the purpose of this was - i thought it was a placeholder insertion so that we could set an array of Object[ ] later.

File: driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java (hunk 2 above)
Comment (chaudum) [inline driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java]: ad 2) leave it up to you

Comment (Autophagy) [inline driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java]: on (1) it turns out that the TypeInfoCache already has a mapping/alias function enabled that already works for some Crate types (like integer -> int4) - think this would suffice?  https://github.com/crate/pgjdbc/blob/crate/pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java#L107 We could map, say, 'string' -> 'varchar' this way and it works properly, without the need for an extra function

Comment (Autophagy) [inline driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java]: on (1) it turns out that the TypeInfoCache already has a mapping/alias function enabled that already works for some Crate types (like integer -> int4) - think this would suffice?  https://github.com/crate/pgjdbc/blob/crate/pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java#L107 We could map, say, 'string' -> 'varchar' this way and it works properly, without the need for an extra function

Comment (chaudum) [inline driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCConnectionTest.java]: yes, we can add "crate aliases" there. please also make a comment about crate aliases in that static block.

File: driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCTypesTest.java [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+    private static Long countUnassigned() throws SQLException {
+        ResultSet rs = connection.createStatement()
+                .executeQuery("SELECT count(*) FROM sys.shards WHERE state != 'STARTED'");
+        rs.next();
+        return rs.getLong(1);
+    }
+
+    @Before
+    public void before() throws SQLException {
+        insertIntoTable();
```
Comment (kovrus) [inline driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCTypesTest.java]: we don't need to insert data into the table before each test, move it to beforeClass

File: driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCTypesTest.java [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+    }
+
+    @After
+    public void after() throws SQLException {
+        deleteFromTable();
```
Comment (kovrus) [inline driver/test/java/io/crate/client/jdbc/integrationtests/CrateJDBCTypesTest.java]: same here
